cutgeneratingfunctionology/spam/parametric_real_field_element.py

```diff
@@ -56,9 +56,9 @@ def __init__(self, parent, value, symbolic=None):
             ## Test coercing the value to RR, so that we do not try to build a ParametricRealFieldElement
             ## from something like a tuple or vector or list or variable of a polynomial ring
             ## or something else that does not make any sense.
-            from sage.interfaces.mathematica import MathematicaElement
-            if not isinstance(value, MathematicaElement):
-                RR(value)
+            # FIXME: parent(value) caused SIGSEGV because of an infinite recursion
+            if not ((type(value) is int) or hasattr(value, 'parent') and RR.has_coerce_map_from(value.parent())):
```

Comment:
`type(value) is int`?? I think this needs a more general solution

Reply:
```
sage: from sage.structure.coerce import py_scalar_parent                                                                                                                                                                                       
sage: py_scalar_parent(type(1r))                                                                                                                                                                                                               
Integer Ring
sage: py_scalar_parent(type([1, 2]))                                                                                                                                                                                                           
```

Reply:
Doesn't work
sage: value = 1                                                                                                 
sage: type(value)                                                                                               
<class 'sage.rings.integer.Integer'>
sage: py_scalar_parent(type(value))

Reply:
value is already an element. this is for values that are not yet